Update alter-database-transact-sql.md (#7639)
MicrosoftDocs/sql-docs: docs/t-sql/statements/alter-database-transact-sql.md

## Overview: Azure Synapse Analytics

> In Azure Synapse Dedicated SQL Pools, `ALTER DATABASE` modifies the maximum size, or service objective for a database.

Reviewer: ```suggestion
In Azure Synapse, `ALTER DATABASE` modifies certain configuration options of a pool.

```

Author: I agree with this comment.

```syntaxsql
ALTER DATABASE { database_name | CURRENT }
{

>   MODIFY ( <edition_option> [, ... n] )

Reviewer: ```suggestion
  MODIFY NAME = new_database_name
| MODIFY ( <edition_option> [, ... n] )
```
Reverting this change as it does apply to dedicated SQL pools formerly SQL DW created via the Azure portal, and we clarify this later in the arguments section.

Author: #sign-off
